.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,249 @@
+name: Run tests
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+  pull_request:
+
+permissions: write-all
+
+env:
+  CONTRACT_DIR: /tmp/neofs-contract
+  LOCODE_DIR: /tmp/neofs-locode-db
+
+jobs:
+  run_tests:
+    runs-on: self-hosted
+    steps:
+    # Cleanup directories before start
+    - name: Cleanup directories before start
+      run: | 
+        rm -rf ${{ env.CONTRACT_DIR }}
+        rm -rf ${{ env.LOCODE_DIR }}
+
+###############################################################################
+###############################################################################
+###############################################################################
+    # Checkout NEOFS parts for the test environment
+    # neofs-testcases
+    - name: Checkout neofs-testcases repository
+      uses: actions/checkout@v3
+      with:
+        repository: nspcc-dev/neofs-testcases
+        ref: ${{ github.event.inputs.neofs_testcases_branch || 'master' }}
```

Comment:
I'm not sure what happens if `neofs_testcases_branch` is specified. What's the result of `neofs_testcases_branch || 'master'` in this case?

Reply:
> I'm not sure what happens if `neofs_testcases_branch` is specified. What's the result of `neofs_testcases_branch || 'master'` in this case?

The value specified in neofs_testcases_branch.
It's not in use now, it's a riddle for the future.
It's probably best to remove this so as not to make the code complicated.